packages/playground/src/pages/repo-settings-rules-page.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useState } from 'react'
+import {
+  Button,
+  ButtonGroup,
+  Icon,
+  StackedList,
+  Input,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  Text,
+  Textarea,
+  Checkbox
+} from '@harnessio/canary'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+
+import { FormFieldSet } from '../index'
+import { MessageTheme } from '../components/form-field-set'
+
+const formSchema = z.object({
+  name: z.string().min(1, 'Name is required'),
+  description: z.string().min(1, 'Description is required'),
+  targetPatterns: z.string().min(1, 'Patterns are required'),
+  includePatterns: z.enum(['', '1', '2', '3']),
+  toggleValue: z.boolean(),
+  bypassValue: z.string().optional(),
+  access: z.enum(['1', '2'], {}),
+  defaultBranchValue: z.boolean().optional(),
+  editPermissionsValue: z.boolean().optional(),
+  rules: z.array(
+    z.object({
+      id: z.string(),
+      checked: z.boolean()
+    })
+  )
+})
+export type FormFields = z.infer<typeof formSchema>
+// interface RepoSettingsRulesPageProps {
+//   onFormSubmit?: (data: FormFields) => void
+//   onFormCancel?: () => void
+//   apiError?: string | null
+//   isLoading?: boolean
+//   isSuccess?: boolean
+// }
```

Comment:
did we need this comment?

Reply:
Yeah will be used when integrating to `gitness`